### PR TITLE
incorporate workflow repository to python

### PR DESCRIPTION
**Pull request by rmarabini**
This pull request incorporates a few lines of code in Scipion so it can connect to the Scipion Workflow Repository.

The workflow repository is available at URL: http://workflows.scipion.i2pc.es/
There is a help page at: https://github.com/I2PC/scipionWorkflowRepository/wiki/Scipion-Workflow-Repository-Help-Page.

No test has been created since I do not know how to test the tk part of Scipion.

You may start the search GUI from Scipion (select project from the menu and then search workflow) or connect directly to the search URL at http://workflows.scipion.i2pc.es. 

To upload a protocol using Scipion select the protocols (click on them while holding the CTRL-key) and then right click and choose upload workflow from the pop up menu. At this point a browser will be opened, fill the form (all fields are compulsory) and press the create button. Thanks for share your workflows ;-).

---------------

Totally unrelated with the above describe problem I also fix a small bug in volume import protocol when a ·D map is imported but not copied and the 3Dmap format is not mrc/ccp4

### CODE REVIEW

File: pyworkflow/gui/project/viewprotocols.py [hunk 1]
```diff
@@ -1826,6 +1833,62 @@ def _export(obj):
...
+            # connection
+            datagen, headers = poster.encode.multipart_encode(params)
+            opener = poster.streaminghttp.register_openers()
+            # create request and connect to server
+            response = opener.open(urllib2.Request(upload_file_url,
+                                   datagen, headers))
+            # server returns dictionary as json with remote name of the saved file
+            _dict = json.loads(response.read())
+            # server has stored file in file named _dict['jsonFileName']
+            # TODO: This version thing is a horrible hack but it is not
```
Comment (pconesa) [inline pyworkflow/gui/project/viewprotocols.py]: Really?

Comment (rmarabini) [reply pyworkflow/gui/project/viewprotocols.py]: I do not know how to answer to the above comment. Should I do something?

Ideally scipion should offer an easy way to get the version something like:

from scipion import version
version = version()

With the current code the only way I know how to get the version is:

+            import imp
 +            currentDir = os.path.dirname(os.path.abspath(__file__))
 +            currentFile = os.path.join(currentDir,'../../../scipion')
 +            mod = imp.load_source('scipion', currentFile)
 +            version = mod.getVersion(False)

the above code is rather ugly

File: pyworkflow/em/protocol/protocol_import/volumes.py [hunk 2]
```diff
@@ -197,8 +201,12 @@ def _methods(self):
                            (self._getVolMessage(), self.samplingRate.get()),)
         return methods
 
-    def _getVolumeFileName(self, fileName):
-        baseFileName="import_" + basename(fileName).split(".")[0] + ".mrc"
+    def _getVolumeFileName(self, fileName, extension=None):
+        if extension is not None: 
+            baseFileName="import_" + basename(fileName).split(".")[0] + ".%s"%extension
```
Comment (delarosatrevin) [inline pyworkflow/em/protocol/protocol_import/volumes.py]: Need some spaces for PEP8

File: pyworkflow/gui/project/constants.py [hunk 3]
```diff
@@ -41,3 +41,8 @@
                STATUS_SCHEDULED: '#F3F5CB'
                }
 
+WORKFLOW_REPOSITORY_SERVER='http://workflows.scipion.i2pc.es/'
```
Comment (delarosatrevin) [inline pyworkflow/gui/project/constants.py]: I don't really like this values in the python code...it should be in some .conf file (and for backward compatibility with some reasonable default)

File: pyworkflow/gui/project/project.py [hunk 4]
```diff
@@ -39,7 +39,7 @@
 import uuid
 import SocketServer
 import tempfile
-
+import webbrowser
```
Comment (delarosatrevin) [inline pyworkflow/gui/project/project.py]: Can this new dependency be moved down to the function where it is used? Or it is pretty much standard and it is not a problem?

File: pyworkflow/gui/project/viewprotocols.py [hunk 5]
```diff
@@ -55,6 +56,9 @@
 
 from constants import STATUS_COLORS
 from pyworkflow.gui.project.utils import getStatusColorFromNode
+import poster.streaminghttp
```
Comment (delarosatrevin) [inline pyworkflow/gui/project/viewprotocols.py]: I don't like spreading more and more dependencies for implementing new functionality. Can this be wrapped into a function? (in another module)

File: pyworkflow/gui/project/viewprotocols.py [hunk 6]
```diff
@@ -1826,6 +1833,62 @@ def _export(obj):
             entryLabel='File', entryValue='workflow.json')
         browser.show()
 
+    def _exportUploadProtocols(self):
```
Comment (delarosatrevin) [inline pyworkflow/gui/project/viewprotocols.py]: Please make sure that no error will happen (or at least that will not interfere with the current processing) in the GUI if case there is not internet connnection or there is any failure

File: pyworkflow/gui/project/viewprotocols.py [hunk 7]
```diff
@@ -1826,6 +1833,62 @@ def _export(obj):
...
+
+        def uploadWorkflow(upload_file_url, upload_metadata_url, jsonFileName):
```
Comment (delarosatrevin) [inline pyworkflow/gui/project/viewprotocols.py]: Maybe this whole function (and its dependencies and imports) can be moved to a separate module / place

**Comment by delarosatrevin** [comment]
@rmarabini I think that maybe we could introduce this  feature for release-1.2.1, what do you think? In my opinion, if we address the dependencies that I pointed out in the comments, it is not much a risk and we could release it earlier.

**Comment by rmarabini** [comment]
OK, let us sit together tomorrow and think what need to be modified

            Roberto

File: pyworkflow/gui/project/constants.py (hunk 3 above)
Comment (delarosatrevin) [reply pyworkflow/gui/project/constants.py]: Or maybe (discussing with @rmarabini ) a config.py file

**Comment by rmarabini** [comment]
I close this pull request since it is going to be backported to Scipion 1.2

**Pull request closed without merging**